src/onnx/parse_pad.cpp

```diff
@@ -169,7 +169,7 @@ struct parse_pad : op_parser<parse_pad>
                                const std::vector<instruction_ref>& args) const
     {
         float value = 0.0f;
-        if(args.size() >= 3 and args.at(2)->get_shape().scalar())
+        if(args.size() >= 3 and args.at(2)->get_shape().scalar() and not args.at(2)->is_undefined())
```

Comment:
The `if clause` (line 172) and the `if clause` (line 180) below: are they testing the same thing? Thanks.

Reply:
No they don't.
An `undefined` shape has has `0` elements thus we want to avoid line 180 if we have `undefined`.
AFAIK scalar shape only checks if the shape is not dynamic or has non-zero stride.

Reply:
Thanks. @pfultz2 I wonder if there should be just a single check for number of elements == 1. No need to check for scalar() here. And that should eliminate the check for is_undefined(), if it has zero elements.

Reply:
@gyulaz-htec I don't want to hold your PR. . It is a bit illogical to first check for the shape of the argument, and then check the argument is not undefined -- the order should be reverse. But.. instead.. 
It is simpler to just check that the number of elements ==1, and just bypass the undefined (and scalar check) here. The error case when elements > 1 can be handled separately (or combined into elements >=1). 
I will approve the PR, but feel feel to include this suggestion if you think it is logical. Thanks.

Reply:
Thanks @lakhinderwalia your suggestion seems more simpler than my solution. I've updated the code according to that.

Reply:
Thanks @gyulaz-htec